src/handlers/pr_tracking.rs

```diff
@@ -0,0 +1,137 @@
+//! This module updates the PR work queue of the Rust project contributors
+//!
+//! Purpose:
+//!
+//! - Adds the PR to the work queue of one or more team member (when the PR has been assigned)
+//! - Removes the PR from the work queue of one or more team members (when the PR is unassigned or closed)
+//!
+//! Notes:
+//!
+//! - When assigning or unassigning a PR, we only receive the list of current assignee(s)
+//! - In case of a complete PR unassignment the list will be empty (i.e. does not have assignee(s) removed)
+//! - The code in this handler MUST be idempotent because GH emits a webhook trigger for every assignee
+
+use crate::{
+    config::ReviewPrefsConfig,
+    db::notifications::record_username,
+    github::{IssuesAction, IssuesEvent},
+    handlers::Context,
+};
+use anyhow::Context as _;
+use itertools::Itertools;
+use tokio_postgres::Client as DbClient;
+use tracing as log;
+
+pub(super) struct ReviewPrefsInput {}
+
+pub(super) async fn parse_input(
+    _ctx: &Context,
+    event: &IssuesEvent,
+    config: Option<&ReviewPrefsConfig>,
+) -> Result<Option<ReviewPrefsInput>, String> {
+    // NOTE: this config check MUST exist. Else, the triagebot will emit an error
+    // about this feature not being enabled
+    if config.is_none() {
+        return Ok(None);
+    };
+
+    // Execute this handler only if this is a PR
+    // and if the action is an assignment or unassignment
+    if !event.issue.is_pr()
+        || !matches!(
+            event.action,
+            IssuesAction::Assigned | IssuesAction::Unassigned
+        )
+    {
+        return Ok(None);
+    }
+    Ok(Some(ReviewPrefsInput {}))
+}
+
+pub(super) async fn handle_input<'a>(
```

Comment:
I didn't find another way around doing this other than:
- removing the PR from everyone
- adding the PR again to the assignee(s) - if any

We need to handle ~three~ four cases:
- PR is unassigned from everyone
- PR is unassigned to one of possibly many assignee(s) - unlikely but allowed by the API
- PR is assigned to one person
- PR is assigned to multiple assignees at the same time - unlikely but allowed by the API